pkg/platform/kube/functionres/lazy.go

```diff
@@ -458,50 +460,41 @@ func (lc *lazyClient) deleteRemovedCronTriggersCronJob(functionLabels labels.Set
 	function *nuclioio.NuclioFunction,
 	newCronTriggers map[string]functionconfig.Trigger) error {
 
-	// list existing cron trigger cron jobs
-	existingCronJobs, err := lc.kubeClientSet.BatchV1beta1().CronJobs(function.Namespace).List(metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("nuclio.io/function-name=%s,nuclio.io/function-cron-trigger-cron-job=true", function.Name),
-	})
-	if err != nil {
-		return errors.Wrap(err, "Failed to list cron jobs")
-	}
-
 	// make a list of all the new cron trigger cron job names
-	newCronJobNames := []string{}
+	var newCronTriggerNames []string
 	for newCronTriggerName := range newCronTriggers {
-		newCronJobNames = append(newCronJobNames, kube.CronJobNameFromFunctionName(function.Name, newCronTriggerName))
+		newCronTriggerNames = append(newCronTriggerNames, newCronTriggerName)
 	}
 
-	// get each cron job that is not inside the new cron triggers and should be deleted
-	cronJobsToDelete := []string{}
-	for _, existingCronJob := range existingCronJobs.Items {
-		if common.StringSliceContainsString(newCronJobNames, existingCronJob.Name) {
-
-			// means that this existing cron job is still inside the function's cron triggers (shall not be deleted)
-			continue
-		}
-
-		cronJobsToDelete = append(cronJobsToDelete, existingCronJob.Name)
+	// retrieve all the cron jobs that aren't inside the new cron triggers, so they can be deleted
+	cronJobsToDelete, err := lc.kubeClientSet.BatchV1beta1().CronJobs(function.Namespace).List(metav1.ListOptions{
+		LabelSelector: lc.compileCronJobLabelSelector(function.Name,
+			fmt.Sprintf("nuclio.io/function-cron-trigger-name notin (%s)",
+				strings.Join(newCronTriggerNames, ", "))),
```

Comment:
Suggest to compile the extraTrigger before the call to `compileCronJobLabelSelector` to keep this clean